app/scripts/modules/core/src/application/nav/NavCategory.tsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+
+import { DataSourceNotifications } from 'core/entityTag/notifications/DataSourceNotifications';
+import { Icon, useDataSource } from '../../presentation';
+
+import { ApplicationDataSource } from '../service/applicationDataSource';
+import { Application } from '../application.model';
+import { IEntityTags } from '../../domain';
+
+export interface INavCategoryProps {
+  category: ApplicationDataSource;
+  isActive: boolean;
+  app: Application;
+}
+
+export const NavCategory = ({ app, category, isActive }: INavCategoryProps) => {
+  const { alerts, badge, iconName, key, label } = category;
+
+  const { data: badgeData } = useDataSource(app.getDataSource(badge || key));
+  const runningCount = badge ? badgeData.length : 0;
+
+  // useDataSource is enough to update alerts when needed
+  useDataSource(category);
```

Comment:
I understand why this works but wonder if it's worth making this more clear/explicit, i.e. pulling alerts off of the result of useDataSource instead of relying knowing (based on the comment) that this is what causes a re-render when that thing over there changes.

Reply:
Yes I get that. I tried a few options before ending on this because the behavior I observed was a bit strange. The result of use `dataSource` updates the `entityTags`.  When I pulled `alerts` directly off of the result of `useDataSource` the alerts would not update, and I would instead have to map/filter `entityTags` to alerts. I ended up here because it was simplest to cause the re-render. 

I can either go back to this approach with the entityTags, or make a more detailed comment to explain the choice. What do you think?

Reply:
If it didn't work I'm fine with it as is.

Separately, I would like us to think about what usage of the `useDataSource` hook looks like longer term though, wrt things like `alerts` that are exposed on the `ApplicationDataSource` itself but not through the hook.

Just feels awkward to be able to use the hook unless you need any of these other things, then still use the hook, but still pull it off the datasource directly.

Reply:
+1 to what @alanmquach said wrt revisiting this pattern

Reply:
Sounds good. Its more of a `subscribeToDataSource` rather than a `useDataSource`